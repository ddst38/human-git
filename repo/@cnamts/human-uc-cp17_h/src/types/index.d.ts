// Here you can define types that
// you'll be using across your application
export interface IndexedObject<T = string> {
	[key: string]: T;
}

export interface User {
    value: Value;
}

export interface Value {
    authorities: Authority[];
    username:    string;
}

export interface Authority {
    authority:  string;
    perimeters: string[];
}
