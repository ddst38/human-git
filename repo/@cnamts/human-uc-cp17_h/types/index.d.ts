import { PluginFunction } from 'vue';

/* eslint-disable @typescript-eslint/no-empty-interface */
export interface InstallationAutoDesComposantsOptions {}

export interface InstallPlugin {
	install: PluginFunction<InstallationAutoDesComposantsOptions>;
}

declare const InstallationAutoDesComposants: InstallPlugin;
export default InstallationAutoDesComposants;
