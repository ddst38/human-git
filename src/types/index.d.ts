// Here you can define types that
// you'll be using across your application

export interface Link {
	id: number;
	url: string;
	name: string;
}

export interface LinkItem {
	title: string;
	links: Link[];
}

export interface User {
    value: Value;
}

export interface Value {
    authorities: Authority[];
    username:    string;
}

export interface Authority {
    authority:  string;
    perimeters: string[];
}
