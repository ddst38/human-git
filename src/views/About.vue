<template>
	<PageContainer>
		<VCard class="pa-8">
			<VBtn
				:to="{
					name: 'home'
				}"
				color="accent"
				class="mb-8"
				outlined
				exact
				data-cy="backBtn"
			>
				<VIcon class="mr-2">
					{{ backArrowIcon }}
				</VIcon>

				{{ $t('views.about.backBtn.label') }}
			</VBtn>

			<h2 class="text-h6 font-weight-bold primary--text">
				{{ $t('views.about.title') }}
			</h2>

			<LinksList
				v-for="(data, index) in links"
				:key="index"
				:title="data.title"
				:links="data.links"
				class="mt-4"
				data-cy="links"
			/>
		</VCard>
	</PageContainer>
</template>

<script lang="ts">
	import Vue from 'vue';
	import Component from 'vue-class-component';

	import { Meta, MetaInfo } from '@/decorators';

	import { LinkItem } from '@/types';

	import { mdiArrowLeft } from '@mdi/js';

	@Component({
		components: {
			LinksList: () => import(
				/* webpackChunkName: "links-list" */
				'@/components/LinksList'
			)
		}
	})
	export default class About extends Vue {
		backArrowIcon = mdiArrowLeft;

		get links(): LinkItem[] {
			return this.$t('views.about.links') as unknown as LinkItem[];
		}

		/* istanbul ignore next */
		@Meta
		metaInfo(): MetaInfo {
			return {
				title: this.$t('views.about.meta.title') as string,
				meta: [
					{
						name: 'description',
						vmid: 'description',
						content: this.$t('views.about.meta.description') as string
					}
				]
			};
		}
	}
</script>
