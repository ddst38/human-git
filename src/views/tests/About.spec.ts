import Vue from 'vue';
import Vuetify from 'vuetify';

import {
	Wrapper,
	shallowMount,
	createLocalVue,
	createVuetifyInstance,
	installGlobalPlugins,
	mockTranslations
} from '@cnamts/vue-dot/src/helpers/testUtils';

import About from '../About.vue';

import { LinkItem } from '@/types';

const links: LinkItem[] = [
	{
		title: 'Test',
		links: [
			{
				id: 0,
				url: 'https://example.com/',
				name: 'Test website'
			}
		]
	}
];

describe('About', () => {
	const localVue = createLocalVue();

	let wrapper: Wrapper<Vue>;
	let vuetify: Vuetify;

	installGlobalPlugins(localVue);

	beforeEach(() => {
		vuetify = createVuetifyInstance();
	});

	it('renders correctly', () => {
		wrapper = shallowMount(About, {
			localVue,
			vuetify,
			mocks: mockTranslations<LinkItem[]>({
				'views.about.links': links
			})
		});

		expect(wrapper).toMatchSnapshot();
	});
});
