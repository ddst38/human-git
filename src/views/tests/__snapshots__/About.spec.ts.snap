// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<pagecontainer-stub size="xl" color="transparent">
  <vcard-stub loaderheight="4" tag="div" class="pa-8">
    <vbtn-stub color="accent" outlined="true" tag="button" activeclass="" exact="true" to="[object Object]" type="button" data-cy="backBtn" class="mb-8">
      <vicon-stub class="mr-2">
        M20,11V13H8L13.5,18.5L12.08,19.92L4.16,12L12.08,4.08L13.5,5.5L8,11H20Z
      </vicon-stub>

      views.about.backBtn.label
    </vbtn-stub>
    <h2 class="text-h6 font-weight-bold primary--text">
      views.about.title
    </h2>
    <!---->
  </vcard-stub>
</pagecontainer-stub>
`;
